### DISCUSSION TITLE
What happens with `vulnerabilityAlerts` when Dependabot alerts dismisses low impact alerts?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#22078] [Suggest an Idea]
### Type of discussion.

I just want to chat

### Tell us more.

Do we need to change anything in our code to account for a user that allows Dependabot alerts to **Dismiss low impact alerts**? I think we rely on the behavior of Dependabot alerts for our `vulnerabilityAlerts` feature?

Quote from our `vulnerabilityAlerts` docs: [^docs]

> Renovate can read GitHub's Vulnerability Alerts to customize its Pull Requests. For this to work, you must enable the [Dependency graph](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#enabling-the-dependency-graph), and [Dependabot alerts](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-security-and-analysis-settings-for-your-repository). [snip]

GitHub users can opt-in to a beta feature for Dependabot alerts, called **Dismiss low impact alerts**. This is currently only supported for npm according to the GitHub Docs. [^gh-docs] Here's a picture:

![image](https://github.com/renovatebot/renovate/assets/34918129/1686d5b5-d615-4654-a7ef-bba75a0c3a1a)

[^gh-docs]: [GitHub Docs, using alert rules to prioritize Dependabot alerts](https://docs.github.com/en/code-security/dependabot/dependabot-alerts/using-alert-rules-to-prioritize-dependabot-alerts)
[^docs]: [`vulnerabilityAlerts` config option docs](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts)

**Comment by rarkins** [comment]
We already have logic to skip over manually dismissed alerts. If they've used the same fields then it might work out of the box, otherwise we'd need to inspect the API definition or have a way to test it live.

**Reply by HonkingGoose** [reply]
Found this relevant quote:

> Auto-dismissed alerts are defined by the `resolution:auto-dismiss` close reason. Automatic dismissal activity is included in alert webhooks, REST and GraphQL APIs, and the audit log. For more information, see "[Dependabot alerts](https://docs.github.com/en/rest/dependabot/alerts)" in the REST API documentation, and the "`repository_vulnerability_alert`" section in "[Reviewing the audit log for your organization](https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/reviewing-the-audit-log-for-your-organization#repository_vulnerability_alert-category-actions)."

Here's the link to the REST API docs for the Dependabot alerts feature: https://docs.github.com/en/rest/dependabot/alerts
